ts/src/ellipx.ts

```diff
@@ -0,0 +1,1583 @@
+                        ],
+                        'taker': [
+                            [ this.parseNumber ('0'), this.parseNumber ('0.0030') ],     // 0-10k: 30bps
+                            [ this.parseNumber ('10000'), this.parseNumber ('0.0025') ], // 10k-50k: 25bps
+                            [ this.parseNumber ('50000'), this.parseNumber ('0.0020') ], // 50k-100k: 20bps
+                            [ this.parseNumber ('100000'), this.parseNumber ('0.0015') ], // 100k-1M: 15bps
+                            [ this.parseNumber ('1000000'), this.parseNumber ('0.0012') ], // 1M-5M: 12bps
+                            [ this.parseNumber ('5000000'), this.parseNumber ('0.0010') ], // 5M-15M: 10bps
+                            [ this.parseNumber ('15000000'), this.parseNumber ('0.0008') ], // 15M-75M: 8bps
+                            [ this.parseNumber ('75000000'), this.parseNumber ('0.0005') ], // 75M-100M: 5bps
+                            [ this.parseNumber ('100000000'), this.parseNumber ('0.0003') ], // 100M+: 3bps
+                        ],
+                    },
+                },
+                'stablecoin': {
+                    'tierBased': false,
+                    'percentage': true,
+                    'maker': this.parseNumber ('0.0000'),  // 0%
+                    'taker': this.parseNumber ('0.000015'), // 0.0015%
+                },
+            },
+            'options': {
+                'defaultType': 'spot',
+                'recvWindow': 5 * 1000,
+                'broker': 'CCXT',
+                'networks': {
+                    'Bitcoin': 'Bitcoin',
+                    'Ethereum': 'ERC20',
+                },
+                'defaultNetwork': 'defaultNetwork',
+                'defaultNetworkCodeReplacements': {
+                    'BTC': 'Bitcoin',
+                    'ETH': 'Ethereum',
+                },
+            },
+            'commonCurrencies': {},
+            'exceptions': {
+                'exact': {
+                    // todo
+                    '400': BadRequest,
+                    '401': AuthenticationError,
+                    '403': PermissionDenied,
+                    '404': BadRequest,
+                    '429': DDoSProtection,
+                    '418': PermissionDenied,
+                    '500': ExchangeError,
+                    '504': ExchangeError,
+                },
+                'broad': {},
+            },
+            'precisionMode': TICK_SIZE,
+        });
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        path = this.implodeParams (path, params);
+        let url = this.urls['api'][api] + '/' + path;
+        if (api === 'private') {
+            this.checkRequiredCredentials ();
+            const nonce = this.uuid ();
+            const timestamp = this.seconds ().toString ();
+            if (method === 'GET') {
+                body = '';
```

Comment:
we don't need to set the body for get requests

Reply:
This is because of according to ellipx docs, it need an empty body hash to query data in get endpoint. As its state in https://docs.google.com/document/d/1ZXzTQYffKE_EglTaKptxGQERRnunuLHEMmar7VC9syM/edit?tab=t.0#heading=h.c23l61oyx6e6 <- "If the request has no body (GET, etc) this needs to be the SHA256 of an empty string."